### PR TITLE
Issue #127 Only sort assoc arrays when creating JSON

### PR DESCRIPTION
**Pull request by andrewmadden**
* Added function to check if array is associative or seq.
* Check if array is associative before sorting for SEB-JSON object.
* Add unit tests to ensure xml elements are serialized correctly.

### CODE REVIEW

File: classes/property_list.php [hunk 1]
```diff
@@ -347,4 +350,17 @@ private function array_serialize_cftypes(array $array) : array {
...
     }
+
+    /**
+     * Check if an array is associative or sequential.
+     *
+     * @param array $array Array to check.
+     * @return bool False if not associative.
+     */
+    private function is_associative_array(array $array) {
+        if (array() === $array) {
```
Comment (dmitriim) [inline classes/property_list.php]: What are you checking here?

File: classes/property_list.php [hunk 2]
```diff
@@ -347,4 +350,17 @@ private function array_serialize_cftypes(array $array) : array {
...
+            return false;
+        }
+        return array_keys($array) !== range(0, count($array) - 1);
```
Comment (dmitriim) [inline classes/property_list.php:365]: as there is no unit tests to cover this complicated line. Can you please explain what's happening here?

Comment (andrewmadden) [reply classes/property_list.php:365]: It checks that the array keys are 0,1,2,3...array_length - 1.

File: classes/property_list.php (hunk 1 above)
Comment (andrewmadden) [reply classes/property_list.php]: It just checks if array is empty. Stole the code. I'll fix it to be more readable

**Pull request merged**